pkg/policies/security/security.go

```diff
@@ -0,0 +1,206 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package security implements the SECURITY.md security policy.
+package security
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"path"
+
+	"github.com/ossf/allstar/pkg/config"
+	"github.com/ossf/allstar/pkg/config/operator"
+	"github.com/ossf/allstar/pkg/policydef"
+
+	"github.com/google/go-github/v32/github"
+	"github.com/rs/zerolog/log"
+)
+
+const configFile = "security.yaml"
+const polName = "SECURITY.md"
+const filePath = "SECURITY.md"
+
+// OrgConfig is the org-level config definition for Branch Protection.
+type OrgConfig struct {
+	// OptConfig is the standard org-level opt in/out config, RepoOverride applies to all
+	// BP config.
+	OptConfig config.OrgOptConfig `yaml:"optConfig"`
+
+	// Action defines which action to take, default log, other: issue...
+	Action string `yaml:"action"`
+
+	//TODO add default contents for "fix" action
+}
+
+// RepoConfig is the repo-level config for Branch Protection
+type RepoConfig struct {
+	// OptConfig is the standard repo-level opt in/out config.
+	OptConfig config.RepoOptConfig `yaml:"optConfig"`
+
+	// Action overrides the same setting in org-level, only if present.
+	Action *string `yaml:"action"`
+}
+
+type mergedConfig struct {
+	Action string
+}
+
+type details struct {
+	Exists bool
+	Empty  bool
+}
+
+var configFetchConfig func(context.Context, *github.Client, string, string, string, interface{}) error
+
+func init() {
+	configFetchConfig = config.FetchConfig
+}
+
+// Security is the SECURITY.md policy object, implements policydef.Policy.
+type Security bool
+
+// NewSecurity returns a new SECURITY.md policy.
+func NewSecurity() policydef.Policy {
+	var s Security
+	return s
+}
+
+// Name returns the name of this policy, implementing policydef.Policy.Name()
+func (s Security) Name() string {
+	return polName
+}
+
+type repositories interface {
+	GetContents(context.Context, string, string, string,
+		*github.RepositoryContentGetOptions) (*github.RepositoryContent,
+		[]*github.RepositoryContent, *github.Response, error)
+}
+
+// Check performs the polcy check for SECURITY.md policy based on the
+// configuration stored in the org/repo, implementing policydef.Policy.Check()
+func (s Security) Check(ctx context.Context, c *github.Client, owner,
+	repo string) (*policydef.Result, error) {
+	return check(ctx, c.Repositories, c, owner, repo)
+}
+
+func check(ctx context.Context, rep repositories, c *github.Client, owner,
+	repo string) (*policydef.Result, error) {
+
+	oc, rc := getConfig(ctx, c, owner, repo)
+	enabled := config.IsEnabled(oc.OptConfig, rc.OptConfig, repo)
+	log.Info().
+		Str("org", owner).
+		Str("repo", repo).
+		Str("area", polName).
+		Bool("enabled", enabled).
+		Msg("Check repo enabled")
+
+	fc, _, rsp, err := rep.GetContents(ctx, owner, repo, filePath, nil)
+	if err != nil {
+		if rsp != nil && rsp.StatusCode == http.StatusNotFound {
```

Comment:
Should this be outside `if err != nil` block? If `err != nil` the response returned should be invalid/irrelevant and to be ignored?